tests/src/Context/TaxonomyContext.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\TaxonomyContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+
+/**
+ * Context with Taxonomy functionality.
+ */
+class TaxonomyContext implements Context {
+
+  /**
+   * List of Vocabularies created during test execution.
+   *
+   * @var \Vocabulary[]
+   */
+  protected $vocabularies = [];
+
+  /**
+   * List of Fields created during test execution.
+   *
+   * @var \Fields[]
+   */
+  protected $fields = [];
+
+  /**
+   * Create vocabulary.
+   *
+   * @param string $name
+   *    Name of the taxonomy.
+   *
+   * @When I create a new vocabulary :name
+   */
+  public function iCreateNewVocabulary($name) {
+    $vocabulary = array(
+      'name' => $name,
+      'machine_name' => $name,
+      'description' => '',
+      'module' => 'taxonomy',
+    );
+    taxonomy_vocabulary_save((object) $vocabulary);
+
+    // Save the vocabulary for cleaning.
+    $this->vocabularies[] = $name;
+  }
+
+  /**
+   * Revert to previous settings after scenario execution.
+   *
+   * @AfterScenario
+   */
+  public function removeVocabularies() {
+    // Remove the vocabularies.
+    foreach ($this->vocabularies as $vocabulary_name) {
+      $vid = db_query("SELECT vid FROM {taxonomy_vocabulary} WHERE machine_name = '$vocabulary_name'")->fetchField();
```

Comment:
I would user taxonomy functions in order to retrieve the vid like:
$vid = taxonomy_vocabulary_machine_name_load('your_vocabulary_name')->vid;
If it is too heavy, I suggest at least to use db_select instead of db_query.